### ISSUE TITLE
Dataframe.reset_index() fails when index is named

### ISSUE THREAD

**Issue by gsakkis** [dask/dask#4454]
```
import numpy as np
import pandas as pd
import dask.dataframe as dd


def test(idx_name=None):
    index = pd.Series(list('abcde'), name=idx_name)
    df = pd.DataFrame({
        'A': np.arange(5, dtype=np.int32),
        'B': np.arange(10, 15, dtype=np.int32),
    }, index=index)
    df = dd.from_pandas(df, npartitions=2)
    df = df.map_partitions(lambda p: (1.0 / p.sum(axis=1)).astype(np.float32),
                           meta=('new', np.float32))
    df = df.reset_index()
    df = df.compute()
    return df


if __name__ == '__main__':
    for idx_name in None, 'index', 'foo':
        print("\nidx_name: {!r}".format(idx_name))
        try:
            print(test(idx_name))
        except ValueError as ex:
            print(ex)
```
```

idx_name: None
  index       new
0     a  0.100000
1     b  0.083333
2     c  0.071429
0     d  0.062500
1     e  0.055556

idx_name: 'index'
  index       new
0     a  0.100000
1     b  0.083333
2     c  0.071429
0     d  0.062500
1     e  0.055556

idx_name: 'foo'
The columns in the computed data do not match the columns in the provided metadata
```

**Comment by mrocklin** [comment]
If I remove your meta= declaration then things are fine.  It's finding that
the declared metadata is diffferent from the actual results.  In this case
that's probably because you're not specifying the index name in the
metadata but one exists.  If you need to pass meta= then I recommend
passing meta= as a full Series with the index name.

**Comment by gsakkis** [comment]
Indeed, removing `meta` works in this trimmed down example but in the actual case I encountered the issue, it couldn't be inferred automatically.

>  If you need to pass meta= then I recommend passing meta= as a full Series with the index name.

After some trial and error, the thing that worked was `meta=pd.Series([], name='new', dtype=np.float32, index=df.index.head(0))`. If this is indeed the best/simplest way to specify it, I'd say there is room for improvement, if nothing else at least in the documentation about `meta`.

**Comment by mrocklin** [comment]
Usability improvements here would be welcome

**Comment by mrocklin** [comment]
@gsakkis do you have any specific recommendations on how to improve documentation here?  Otherwise I'm going to close this.

**Comment by gsakkis** [comment]
Documentation aside I have the feeling this case could be handled better by inferring the correct index name; will need to look into it and hopefully post a PR.

**Comment by mrocklin** [comment]
Glad to hear it!

**Comment by jcrist** [comment]
This should be fixed by #4509. In the case of metadata passed as a spec (tuple, list of tuples, or dict), we now copy over the index metadata from the first dataframe passed to the function (usually what the user wants).

Running your example:

```bash
$ python test.py

idx_name: None
  index       new
0     a  0.100000
1     b  0.083333
2     c  0.071429
0     d  0.062500
1     e  0.055556

idx_name: 'index'
  index       new
0     a  0.100000
1     b  0.083333
2     c  0.071429
0     d  0.062500
1     e  0.055556

idx_name: 'foo'
  foo       new
0   a  0.100000
1   b  0.083333
2   c  0.071429
0   d  0.062500
1   e  0.055556
```

**Comment by lysanderzealot244** [comment]
> [@gsakkis](https://github.com/gsakkis) do you have any specific recommendations on how to improve documentation here? Otherwise I'm going to close this.





OFZLmcw4cmGw
